Dockerfile.imager

```diff
@@ -0,0 +1,30 @@
+########## Builder ##########
+# Build the imager binary
```

Comment:
I thought we wanted to build only one image?

Reply:
Indeed, we want to / will productize only one image, comprising both LCA and Imager.

The `imager-build-container` target is intended for folks developing / testing **only** the Imager tool, without including the LCA bits in a lighter image.

Reply:
Another idea would be to have another stage in the lifecycle-agent Dockerfile, e.g. :

```dockerfile
FROM registry.access.redhat.com/ubi9/ubi:latest as imager-dev

RUN dnf -y install jq && \
    dnf clean all && \
    rm -rf /var/cache/dnf

COPY --from=imager /opt/app-root/src/build/ibu-imager /usr/local/bin/ibu-imager
COPY ibu-imager/installation_configuration_files/ /usr/local/bin/installation_configuration_files/

ENTRYPOINT ["/usr/local/bin/ibu-imager"]
```

and build the imager-dev image with something like this:

```shell
podman build . -t quay.io/mresvani/ibu-imager -f Dockerfile --target imager-dev
```

This way, we're only building the imager part and we have a dev image, which is small enough (I guess we could go smaller by using ubi-minimal or perhaps ubi-micro?). WDYT?

Reply:
tbh, I don't have a strong preference here, as long as we get a smaller image for testing / developing Imager, I guess both approaches are OK to me.

(just created a separate Dockerfile cause it seemed kind of a different / independent piece of software)

/deferring to @jc-rh @donpenney on this

Reply:
I like Michail's idea. Less chance for us to forget to update the standalone file. And yes, we typically use ubi-minimal for pure go app. Not sure if the imager requires anything additional.

Reply:
The LCA was originally using gcr.io/distroless/static:nonroot, and updated to ubi:latest because that's what the ibu-imager was using. So presumably we just need to ensure any dependencies required by the imager are met. Smaller is definitely better, and less chances of being impacted by CVEs, etc

Reply:
If you're looking to make a lighter image, using a smaller base FROM image would be the way to go, I think. The LCA manager binary isn't much bigger than the ibu-imager binary

Reply:
yea, originally I also tried to use `gcr.io/distroless/static:nonroot` in Imager but had some issues with `nsenter` missing in that base. Let's do Michail's idea, thanks y'all.

Reply:
```
-> podman run --rm gcr.io/distroless/static:nonroot nsenter --version
Error: crun: executable file `nsenter` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found

-> podman run --rm -ti registry.access.redhat.com/ubi9/ubi-micro:latest nsenter --version
Error: crun: executable file `nsenter` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found

-> podman run --rm -ti registry.access.redhat.com/ubi9/ubi-minimal:latest nsenter --version
Error: crun: executable file `nsenter` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found

-> podman run --rm -ti registry.access.redhat.com/ubi9/ubi:latest nsenter --version
nsenter from util-linux 2.37.4
```

Reply:
If you feel you need a separate image, ok. But the only difference between the LCA image and this ibu-imager image is the exclusion of the manager binary, which is currently 53M. So I don't understand what the benefit would be here.